functions/go/gatekeeper/README.md

```diff
@@ -4,29 +4,93 @@
 
 <!--mdtogo:Short-->
 
-Validate the KRM resources using [Gatekeeper] policies.
+The `gatekeeper` function wraps the [`Gatekeeper`] policy enforcement engine to
+evaluate [Gatekeeper] constraints to validate the KRM resources.
```

Comment:
Another wording (heavily inspired by the OPA blogpost https://kubernetes.io/blog/2019/08/06/opa-gatekeeper-policy-and-governance-for-kubernetes/) itself:

The `gatekeeper` function enforces policies on the package resources. You can define policies for governance and legal requirements or to enforce best practices and organizational conventions.
 
For example, you can enforce policies like:
- `ConfigMap` must not contain fields with `private_key` name
- All pods must have resource limits
- All namespaces must have a label that lists a point-of-contact